pilot/pkg/model/jwks_resolver.go

```diff
@@ -186,27 +186,22 @@ func (r *jwksResolver) GetPublicKey(jwksURI string) (string, error) {
 	now := time.Now()
 	if val, found := r.keyEntries.Load(jwksURI); found {
 		e := val.(jwtPubKeyEntry)
-
-		// Return from cache if it's not expired.
-		if e.expireTime.After(now) {
-			// Update cached key's last used time.
-			e.lastUsedTime = now
-			r.keyEntries.Store(jwksURI, e)
-			return e.pubKey, nil
-		}
+		// Update cached key's last used time.
+		e.lastUsedTime = now
+		r.keyEntries.Store(jwksURI, e)
+		return e.pubKey, nil
 	}
 
-	// Fetch key if it's not cached, or cached item is expired.
-	resp, err := r.getRemoteContent(jwksURI)
+	// Fetch key if it's not cached, only retry once as this is in the critical path for pushing configs.
+	resp, err := r.getRemoteContentWithRetry(jwksURI, 1)
```

Comment:
as discussed, safer to don't retry in critical path; (or need network team's review)

Reply:
I set to a very conservative value, only retry once,
@philrud @howardjohn @costinm , could provide some input from the pilot network perspective? Previously (without retry), it will block on this for 5 seconds (HTTP timeout) in the worst case. Now (with retry) the worst case is 11 seconds (5 + 1 + 5). is this okay for pilot?